app/views/cocktails/view.php

```diff
@@ -16,6 +16,14 @@
 $totalLikes = $this->cocktailService->getLikeCount($cocktailId);
 $cocktailTitle = htmlspecialchars($cocktail->getTitle() ?? 'Unknown Cocktail');
 ?>
+<?php if (isset($_SESSION['badge_notification'])): ?>
+    <?php error_log("Displaying badge notification: " . print_r($_SESSION['badge_notification'], true)); ?>
+    <div class="alert alert-success">
+        <h4><?= htmlspecialchars($_SESSION['badge_notification']['message']); ?></h4>
+        <p><?= htmlspecialchars($_SESSION['badge_notification']['badge_description']); ?></p>
+    </div>
+    <?php unset($_SESSION['badge_notification']); ?>
+<?php endif; ?>
```

Comment:
IS there a reason why this is in cocktails/view.php ?

Reply:
After the users share a recipe, they are directed to the recipe they have shared.
It made most sense to display the notification right after completion of their milestone-recipe on the cocktail page.